src/v/cloud_storage/access_time_tracker.cc

```diff
@@ -78,39 +181,35 @@ void access_time_tracker::remove_timestamp(std::string_view key) noexcept {
     }
 }
 
-void access_time_tracker::remove_others(const access_time_tracker& t) {
+ss::future<>
+access_time_tracker::trim(const std::vector<file_list_item>& existent) {
+    absl::btree_set<uint32_t> existent_hashes;
+    for (const auto& i : existent) {
+        existent_hashes.insert(xxhash_32(i.path.data(), i.path.size()));
+    }
+
     table_t tmp;
-    for (auto it : _table.data) {
-        if (t._table.data.contains(it.first)) {
-            tmp.data.insert(it);
+    for (auto it : _table) {
```

Comment:
i'm still a little confused about how it is safe to iterate over _table with an interspersed `co_await yield` and no lock, but other places (like add/remove timestamp) grab a lock to update _table.

Reply:
Good catch, I extended use of the lock to cover this function